src/main/java/team9/ddang/member/service/MemberServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+package team9.ddang.member.service;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import team9.ddang.member.entity.Member;
+import team9.ddang.member.jwt.service.JwtService;
+import team9.ddang.member.repository.MemberRepository;
+import team9.ddang.member.service.request.JoinServiceRequest;
+import team9.ddang.member.service.response.MemberResponse;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+@Slf4j
+public class MemberServiceImpl implements MemberService {
+
+    private final MemberRepository memberRepository;
+    private final JwtService jwtService;
+
+    @Override
+    public MemberResponse join(JoinServiceRequest serviceRequest, HttpServletResponse response) {
+
+        Member member = serviceRequest.toEntity();
+
+        String accessToken = jwtService.createAccessToken(member.getEmail(), member.getProvider().name());
+        String refreshToken = jwtService.createRefreshToken(member.getEmail());
+
+        jwtService.sendAccessAndRefreshToken(response, accessToken, refreshToken);
+        jwtService.saveRefreshTokenToRedis(member.getEmail(), refreshToken);
+
+        memberRepository.save(member);
+
+        return MemberResponse.from(member);
+    }
```

Comment:
토큰을 생성 및 Redis에 저장하는 로직이 있는데, 토큰을 Redis에 저장하고 DB에 member저장을 실패하면 Redis에 남은 토큰이 문제를 발생시킬 것 같습니다.

Redis는 JPA의 영속성 컨텍스트에 포함되지 않기 때문에, 트렌잭션 롤백에 영향을 받지 않습니다.
즉, member가 DB에 저장되지 않았지만, Redis에 refreshToken이 남을 수 있습니다.

memberRepository.save(member)를 먼저 호출하여 데이터베이스에 성공적으로 저장된 후 토큰 생성 및 Redis 저장 로직을 실행하면 이를 방지할 수 있습니다.

```suggestion
    @Override
    public MemberResponse join(JoinServiceRequest serviceRequest, HttpServletResponse response) {

        Member member = serviceRequest.toEntity();
        
        memberRepository.save(member);

        String accessToken = jwtService.createAccessToken(member.getEmail(), member.getProvider().name());
        String refreshToken = jwtService.createRefreshToken(member.getEmail());

        jwtService.sendAccessAndRefreshToken(response, accessToken, refreshToken);
        jwtService.saveRefreshTokenToRedis(member.getEmail(), refreshToken);

        

        return MemberResponse.from(member);
    }
```

Reply:
중요한 부분인데 제가 미처 생각을 못했네요.. 요구사항 반영해서 수정했습니다. 감사합니다!